src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -37,6 +58,6 @@ public CommandResult execute(Model model) {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof FindCommand // instanceof handles nulls
-                && predicate.equals(((FindCommand) other).predicate)); // state check
+                && namePredicate.equals(((FindCommand) other).namePredicate)); // state check
```

Comment:
```suggestion
                && namePredicate.equals(((FindCommand) other).namePredicate)
                && idPredicate.equals(((FindCommand) other).idPredicate)); // state check
```

Reply:
Right now, I think any 2 `FindCommands` using `idPredicate` would be equal since `namePredicate == null` in the 
constructor. 

```
public FindCommand(StudentIdContainsKeywordsPredicate predicate) {
        this.idPredicate = predicate;
        this.namePredicate = null;
    }
```
Adding the `idPredicate` state check would ensure 2 commands are equal.

Reply:
Thanks for the catch @brelkh! I've indeed missed out on updating the equals method. Regarding your suggested changes, I believe that directly calling `equals()` when any of the attributes are null will result in a `NullPointerException` as there are no checks to see if `this.namePredicate` and `this.idPredicate` are not null, and it will result in something like `null.equals(...)`. I will update the method with the corresponding null checks to ensure that does not happen.